main/test/ca/uwaterloo/flix/language/phase/TestNamer.scala

```diff
@@ -901,6 +901,52 @@ class TestNamer extends FunSuite with TestUtils {
     expectError[NameError.DuplicateUpperName](result)
   }
 
+  test("DuplicateUpperName.21") {
+    val input =
+      """
+        |import java.sql.Statement
+        |enum Statement
+        |""".stripMargin
+    val result = compile(input, Options.TestWithLibNix)
+    expectError[NameError.DuplicateUpperName](result)
+  }
+
+  test("DuplicateUpperName.22") {
+    val input =
+      """
+        |enum Statement
+        |type alias Statement = Int
+        |""".stripMargin
+    val result = compile(input, Options.TestWithLibNix)
+    expectError[NameError.DuplicateUpperName](result)
+  }
+
+  // test("DuplicateUpperName.23") {
```

Comment:
Whats the plan for this test?

Reply:
As I mentioned at the top of this PR:

> I think that the right way to address the latter is to merge ImportEnv into UseEnv, but that involves largely undoing the change I made a while ago to plumb ImportEnv through the namer. Given that that means hundreds of lines changing (eugh!) I wanted to check this in separately.

(so, when we add a new upperName to UseEnv, we'll check that it doesn't clash with the imports, and when we add a new import, we'll check that it doesn't clash with an upperName).

Reply:
OK, can you mark the test as ignored then?

Reply:
Instead of commented out, you mean? Sure, once I work out how to do that 🤣

Reply:
I think you just change test(".. to ignore("...

Reply:
I think that this is ready to be merged.